tests/integration/TestTransaction.cc

```diff
@@ -0,0 +1,322 @@
+        consumer.setLogLevel(LOG_CRIT);
+        consumer.subscribe({topic});
+
+        auto records = KafkaTestUtility::ConsumeMessagesUntilTimeout(consumer);
+
+        std::vector<std::string> messages;
+        for (const auto& record: records) {
+            std::cout << "[" << Utility::getCurrentTime() << "] Consumer got message: " << record.toString() << std::endl;
+            messages.emplace_back(record.value().toString());
+        }
+
+        return messages;
+    };
+
+    {
+        std::cout << "========== Producer: commitTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to commitTransaction", topic, TransactionClosureAction::ToComplete);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadCommitted);
+        ASSERT_EQ(1, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: commitTransaction, Consumer: isolation.level=read_uncommitted ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to commitTransaction", topic, TransactionClosureAction::ToComplete);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadUnCommitted);
+        ASSERT_EQ(1, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: abortTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to abortTransaction", topic, TransactionClosureAction::ToAbort);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadCommitted);
+        ASSERT_EQ(0, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: abortTransaction, Consumer: isolation.level=unread_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to abortTransaction", topic, TransactionClosureAction::ToAbort);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadUnCommitted);
+        ASSERT_EQ(0, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: no commit/abortTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message with no commit/abortTransaction", topic, TransactionClosureAction::ToAbort);
```

Comment:
Should this be NoAction instead ToAbort?

Reply:
OH, here should test with `NoAction` as the parameter. 😅